File: app/src/main/kotlin/com/moviereel/data/api/model/BaseResultsResponse.java
package com.moviereel.data.api.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.greenrobot.greendao.annotation.Id;
import org.greenrobot.greendao.annotation.Property;

import java.util.List;

/**
 * @author lusinabrian on 31/03/17
 * Base response
 */

public class BaseResultsResponse {

    @Expose
    @SerializedName("id")
    private int id;

    @Expose
    @SerializedName("adult")
    private boolean adult;

    @Expose
    @SerializedName("poster_path")
    private String posterPath;

    @Expose
    @SerializedName("overview")
    private String overview;

    @Expose
    @SerializedName("original_language")
    private String originalLanguage;

    @Expose
    @SerializedName("backdrop_path")
    private String backdropPath;

    @Expose
    @SerializedName("video")
    private boolean video;

    @Expose
    @SerializedName("vote_count")
    private int voteCount;

    @Expose
    @SerializedName("vote_average")
    private float voteAverage;

    @Expose
    @SerializedName("popularity")
    private float popularity;

    @Expose
    @SerializedName("genre_ids")
    private List<Integer> genreIds;

    public boolean isAdult() {
        return adult;
    }

    public void setAdult(boolean adult) {
        this.adult = adult;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public void setPosterPath(String posterPath) {
        this.posterPath = posterPath;
    }

    public String getOverview() {
        return overview;
    }

    public void setOverview(String overview) {
        this.overview = overview;
    }

    public String getOriginalLanguage() {
        return originalLanguage;
    }

    public void setOriginalLanguage(String originalLanguage) {
        this.originalLanguage = originalLanguage;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public void setBackdropPath(String backdropPath) {
        this.backdropPath = backdropPath;
    }

    public boolean isVideo() {
        return video;
    }

    public void setVideo(boolean video) {
        this.video = video;
    }

    public int getVoteCount() {
        return voteCount;
    }

    public void setVoteCount(int voteCount) {
        this.voteCount = voteCount;
    }

    public float getVoteAverage() {
        return voteAverage;
    }

    public void setVoteAverage(float voteAverage) {
        this.voteAverage = voteAverage;
    }

    public float getPopularity() {
        return popularity;
    }

    public void setPopularity(float popularity) {
        this.popularity = popularity;
    }

    public List<Integer> getGenreIds() {
        return genreIds;
    }

    public void setGenreIds(List<Integer> genreIds) {
        this.genreIds = genreIds;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @Override
    public String toString() {
        return "BaseResultsResponse{" +
                ", adult=" + adult +
                ", posterPath='" + posterPath + '\'' +
                ", overview='" + overview + '\'' +
                ", originalLanguage='" + originalLanguage + '\'' +
                ", backdropPath='" + backdropPath + '\'' +
                ", video=" + video +
                ", voteCount=" + voteCount +
                ", voteAverage=" + voteAverage +
                ", popularity=" + popularity +
                ", genreIds=" + genreIds +
                '}';
    }

}

File: app/src/main/kotlin/com/moviereel/data/api/model/BaseResultsResponse.kt
package com.moviereel.data.api.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.moviereel.data.db.entities.GenreEntity

/**
 * @author lusinabrian on 31/03/17
 * * Base response
 */
sealed class BaseResultsResponse {
    /***/
    data class MovieLatestResponse(
            @Expose
            @SerializedName("belongs_to_collection")
            private val belongsToCollection: String,

            @Expose
            @SerializedName("budget")
            private val budget: Int,

            @Expose
            @SerializedName("homepage")
            private val homepage: String,

            @Expose
            @SerializedName("genres")
            private val genreResponse: GenreEntity,

            @Expose
            @SerializedName("imdb_id")
            private val imdbId: String,

            @Expose
            @SerializedName("production_companies")
            private val productionCompanies: ArrayList<ProductionCompany>,

            @Expose
            @SerializedName("production_countries")
            private val productionCountries: ArrayList<ProductionCountry>,

            @Expose
            @SerializedName("revenue")
            private val revenue: Int,

            @Expose
            @SerializedName("runtime")
            private val runtime: Int,

            @Expose
            @SerializedName("spoken_languages")
            private val spokenLanguages: ArrayList<SpokenLanguage>,

            @Expose
            @SerializedName("status")
            private val status: String,

            @Expose
            @SerializedName("tagline")
            private val tagline: String
    ) : BaseResultsResponse()

    data class TvResultsResponse(

            @Expose
            @SerializedName("first_air_date")
            var firstAirDate: String,

            @Expose
            @SerializedName("origin_country")
            var originCountry: List<String>,

            @Expose
            @SerializedName("name")
            var name: String,

            @Expose
            @SerializedName("original_name")
            var originalName: String
    ) : BaseResultsResponse()
}
